src/opm/parser/eclipse/EclipseState/Aquancon.cpp

```diff
@@ -55,37 +52,44 @@ namespace Opm {
-            aqurecords.at(aquanconRecordIdx).k1 = aquanconRecord.getItem("K1").template get<int>(0);
-            aqurecords.at(aquanconRecordIdx).k2 = aquanconRecord.getItem("K2").template get<int>(0);
+            // offset the indices
+            const int i1 = aquanconRecord.getItem("I1").template get<int>(0) - 1;
+            const int i2 = aquanconRecord.getItem("I2").template get<int>(0) - 1;
+            const int j1 = aquanconRecord.getItem("J1").template get<int>(0) - 1;
+            const int j2 = aquanconRecord.getItem("J2").template get<int>(0) - 1;
+            const int k1 = aquanconRecord.getItem("K1").template get<int>(0) - 1;
+            const int k2 = aquanconRecord.getItem("K2").template get<int>(0) - 1;
 
-            aquiferID_per_record.at(aquanconRecordIdx) = aquanconRecord.getItem("AQUIFER_ID").template get<int>(0);
             m_maxAquID = (m_maxAquID < aquiferID_per_record.at(aquanconRecordIdx) )?
                             aquiferID_per_record.at(aquanconRecordIdx) : m_maxAquID;
 
-            double influx_mult = aquanconRecord.getItem("INFLUX_MULT").getSIDouble(0);
+            const FaceDir::DirEnum faceDir = FaceDir::FromString(aquanconRecord.getItem("FACE").getTrimmedString(0));
 
-            FaceDir::DirEnum faceDir = FaceDir::FromString(aquanconRecord.getItem("FACE").getTrimmedString(0));
+            // whether allow aquifer connections to locate inside reservoir
+            const std::string& str_inside_reservoir  = aquanconRecord.getItem("CONNECT_ADJOINING_ACTIVE_CELL").getTrimmedString(0);
+            // not sure whether we should give a warning when input other than "YES" or "NO"
+            const bool allow_aquifer_inside_reservoir = str_inside_reservoir == "YES" ? true : false;
 
             // Loop over the cartesian indices to convert to the global grid index
-            for (int k=aqurecords.at(aquanconRecordIdx).k1; k <= aqurecords.at(aquanconRecordIdx).k2; k++) {
-                for (int j=aqurecords.at(aquanconRecordIdx).j1; j <= aqurecords.at(aquanconRecordIdx).j2; j++)
-                    for (int i=aqurecords.at(aquanconRecordIdx).i1; i <= aqurecords.at(aquanconRecordIdx).i2; i++)
-                        aqurecords.at(aquanconRecordIdx).global_index_per_record.push_back
-                                                            (
-                                                                grid.getGlobalIndex(i-1, j-1, k-1)
-                                                            );
+            for (int k = k1; k <= k2; k++) {
+                for (int j = j1; j <= j2; j++) {
+                    for (int i = i1; i <= i2; i++) {
+                        if ( grid.cellActive(i, j, k) ) { // the cell itself needs to be active
+                            if (  allow_aquifer_inside_reservoir
+                               || !neighborCellInsideReservoir(grid, i, j, k, faceDir) ) {
+                                aqurecords.at(aquanconRecordIdx)
```

Comment:
Any reason to not use `[..]`?

Reply:
It is the old code not changed. I will address.